homeassistant/components/mqtt/lock.py

```diff
@@ -178,3 +192,20 @@ async def async_unlock(self, **kwargs):
             # Optimistically assume that the lock has changed state.
             self._state = False
             self.async_write_ha_state()
+
+    async def async_open(self, **kwargs):
+        """Open the door latch.
+
+        This method is a coroutine.
+        """
+        mqtt.async_publish(
+            self.hass,
+            self._config[CONF_COMMAND_TOPIC],
+            self._config[CONF_PAYLOAD_OPEN],
+            self._config[CONF_QOS],
+            self._config[CONF_RETAIN],
+        )
+        if self._optimistic:
+            # Optimistically assume that the lock has changed state.
+            self._state = False
+            self.async_write_ha_state()
```

Comment:
Why does the lock change states to "unlocked" when the latch opens? Is that a valid assumption?

Reply:
![image](https://user-images.githubusercontent.com/12692680/143144066-94b8ce43-634d-41a4-aa28-b463f036a067.png)

The lock can assume 3 states
- Riegel/bolt is locked + Falle/latch is locked
- Riegel/bolt is unlocked + Falle/latch is locked
- Riegel/bolt is unlocked + Falle/latch is unlocked --> Door opens automatically through the seal

The state where the bolt is locked, but the latch is unlocked can be done manually, but not automatically. The automatic door opener always has to open the bolt before it can open the latch. This is mechanically locked.
With this state, the command "open" always causes the latch and bolt to be opened. Where, on the other hand, the command "unlock" only leads to opening the bolt. The latch is in this case is always locked.

Reply:
That depends on whether you look at it from the perspective of the door or the lock. If we remove the `STATE_OPEN`, then the resulting state of the lock is `unlocked` (the state of the door id `open`).